src/renderers/dom/server/ReactServerUpdateQueue.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactServerUpdateQueue
+ * @flow
+ */
+
+'use strict';
+
+var ReactUpdateQueue = require('ReactUpdateQueue');
+var Transaction = require('Transaction');
+var ReactClass = require('ReactClass');
+var warning = require('warning');
+
+function warnNoop(publicInstance: ReactClass, callerName: string) {
+  if (__DEV__) {
+    var instConstructor = publicInstance.constructor;
+    warning(
+      false,
+      '%s(...): Can only update a mounting component. ' +
+      'This usually means you called %s() outside componentWillMount() on the server. ' +
+      'This is a no-op. Please check the code for the %s component.',
+      callerName,
+      callerName,
+      instConstructor && (instConstructor.displayName || instConstructor.name) || 'ReactClass'
+    );
+  }
+}
+
+/**
+ * This is the server update queue. It uses ReactUpdateQueue while mounting
+ * then switches to ReactNoopUpdateQueue when mounted
+ * @class ReactServerUpdateQueue
+ * @param {Transaction} transaction
+ */
+class ReactServerUpdateQueue {
+  /* :: transaction: Transaction; */
+
+  constructor(transaction: Transaction) {
+    this.transaction = transaction;
+  }
+
+  /**
+   * Checks whether or not this composite component is mounted.
+   * @param {ReactClass} publicInstance The instance we want to test.
+   * @return {boolean} True if mounted, false otherwise.
+   * @protected
+   * @final
+   */
+  isMounted(publicInstance: ReactClass): boolean {
```

Comment:
May I propose something else ? Maybe we should use flow's public react interface types there as well: such as React.Component. Or just use a mixed type in the meantime given the fact that mixed is generic enough to avoid problems